lib/cachedir.js

```diff
@@ -64,33 +87,31 @@ Cachedir.prototype.download = function(url, force, callback) {
               if (err) {
                 log.trace('couldn\'t lock ' + output_file + ', seems like the file is still being downloaded (' + err.message + ')');
                 return setTimeout(function() {
-                  self.download(url, force, callback);
+                  self.download(url, force, finish);
                 }, 0); // retry
               }
 
               log.verbose('the file "' + path.basename(output_file) + '" is already downloaded, skipping...');
-              callback(null, output_file);
+              finish(null, output_file);
             });
           });
         });
       } else {
         mkdirp(path.dirname(output_file), function(err) {
-          if (err) return callback(new Error(err));
+          if (err) return finish(new Error(err));
 
           // prepare the writer for piping
-          var writer = fs.createWriteStream(output_file, {
-              flags: 'a'
```

Comment:
The default mode flag is 'w', not 'a'ppend. With 'a', a force download for an asset was redownloading, but appending to the tail of whatever file was already there - totally broken. Now each download should truncate file to zero length and then pipe new contents into it.